azure-pipelines.yml

```diff
@@ -46,13 +48,130 @@ steps:
+            "OperationCode": "SigntoolSign",
+            "Parameters": {
+                "OpusName": "Microsoft",
+                "OpusInfo": "http://www.microsoft.com",
+                "Append": "/as",
+                "FileDigest": "/fd \"SHA256\"",
+                "PageHash": "/NPH",
+                "TimeStamp": "/tr \"http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer\" /td sha256"
+            },
+            "ToolName": "sign",
+            "ToolVersion": "1.0"
+        },
+        {
+            "KeyCode": "CP-231522",
+            "OperationCode": "SigntoolVerify",
+            "Parameters": {},
+            "ToolName": "sign",
+            "ToolVersion": "1.0"
+        }
+      ]
+    SessionTimeout: '60'
+    MaxConcurrency: '50'
+    MaxRetryAttempts: '5'
+  condition: and(succeeded(),or (eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['Build.SourceBranch'], 'refs/heads/release/3.0')))
+- pwsh: |
+    .\repackageBinaries.ps1
+  displayName: Sign and repackage binaries
+  env:
+    AzureBlobSigningConnectionString: $(AzureBlobSigningConnectionString)
+    BuildArtifactsStorage: $(BuildArtifactsStorage)
+    DURABLE_STORAGE_CONNECTION: $(DURABLE_STORAGE_CONNECTION)
+    TELEMETRY_INSTRUMENTATION_KEY: $(TELEMETRY_INSTRUMENTATION_KEY)
+  condition: and(succeeded(),or (eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['Build.SourceBranch'], 'refs/heads/release/3.0')))
+- task: DotNetCoreCLI@2
+  inputs:
+    command: 'run'
+    workingDirectory: '.\build'
+    arguments: 'TestSignedArtifacts --signTest'
+  displayName: 'Verify signed binaries'
+  condition: and(succeeded(),or (eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['Build.SourceBranch'], 'refs/heads/release/3.0')))
+- pwsh: |
+    .\generateMsiFiles.ps1 -MsiGenBranches @("v3.x") -CommitMessage '$(Build.SourceVersionMessage)'
+  env:
+    AzureBlobSigningConnectionString: $(AzureBlobSigningConnectionString)
+    BuildArtifactsStorage: $(BuildArtifactsStorage)
+    DURABLE_STORAGE_CONNECTION: $(DURABLE_STORAGE_CONNECTION)
+    TELEMETRY_INSTRUMENTATION_KEY: $(TELEMETRY_INSTRUMENTATION_KEY)
+  displayName: 'Generate MSI files'
+- task: PowerShell@2
+  inputs:
+    targetType: 'inline'
+    script: |
+      function GenerateSha([string]$filePath,[string]$artifactsPath, [string]$shaFileName)
+      {
+        $sha = (Get-FileHash $filePath).Hash.ToLower()
+        $shaPath = Join-Path $artifactsPath "$shaFileName.sha"
+        Out-File -InputObject $sha -Encoding ascii -FilePath $shaPath
+      }
+
+      Set-Location ".\build"
+      
+      $artifactsPath = Resolve-Path "..\artifacts\"
+      $zipFilesSearchPath = Join-Path $artifactsPath "*.zip"
+      $zipFiles  = Get-ChildItem -File $zipFilesSearchPath
+      
+      foreach($zipFile in $zipFiles)
+      {
+          $zipFullPath = $zipFile.FullName
+          $fileName = $zipFile.Name
+          GenerateSha $zipFullPath $artifactsPath $fileName
+      }
```

Comment:
I'd much rather have these be in their own scripts that get called inline. It makes the yaml a lot easier to read.